File: Leetcode/Leetcode_739.java
// 每日温度
class Solution {
    public int[] dailyTemperatures(int[] T) {
        int n = T.length;
        int[] res = new int[n];
        Deque<Integer> stk = new LinkedList<Integer>();
        for (int i = n - 1; i >= 0; --i) {
            while (!stk.isEmpty() && T[stk.peek()] <= T[i]) {
                stk.pop();
            }
            if(stk.isEmpty()) res[i] = 0;
            else res[i] = stk.peek() - i;
            stk.push(i);
        }
        return res;
    }
}

File: Leetcode/Leetcode_739.cpp
// 每日温度
// 辅助栈
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        int n = T.size();
        vector<int> res(n);
        stack<int> s;
        for(int i = 0; i < n; ++i) {
            while(!s.empty() && T[i] > T[s.top()]) {
                int temp = s.top();
                res[temp] = i - temp;
                s.pop();
            }
            s.push(i);
        }
        return res;
    }
};